openedx/features/course_experience/tests/test_course_updates.py

```diff
@@ -0,0 +1,121 @@
+"""
+Tests for the course updates utility methods.
+"""
+
+from django.test.client import RequestFactory
+
+from openedx.core.djangoapps.user_api.course_tag.api import get_course_tag, set_course_tag
+from openedx.features.course_experience.course_updates import (
+    dismiss_current_update_for_user, get_current_update_for_user, get_ordered_updates,
+)
+from openedx.features.course_experience.tests import CourseUpdatesTestCase
+
+
+class TestCourseUpdatesUtils(CourseUpdatesTestCase):
+    """Tests for the course update utility methods."""
+
+    UPDATES_TAG = 'view-welcome-message'
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.request = RequestFactory().get('/')
+        cls.request.user = cls.user
+
+    def test_update_structure(self):
+        """Test that returned item dictionary is as we expect."""
+        content = '<em>HTML Content</em>'
+        date = 'January 1, 2000'
+        self.create_course_update(content, date=date)
+        updates = get_ordered_updates(self.request, self.course)
+        self.assertListEqual(updates, [{
+            'id': 1,
+            'content': content,
+            'date': date,
+            'status': 'visible',
+        }])
+
+    def test_ordered_updates(self):
+        """Test that order of returned items follows our rules."""
+        first = self.create_course_update('2000', date='January 1, 2000')
+        second = self.create_course_update('2017', date='January 1, 2017')
+        third = self.create_course_update('Also 2017', date='January 1, 2017')
```

Comment:
Same date updates. How do those get sorted? Just default order they show up in would be my guess?

Reply:
Date first, then order they were created.